app/src/test/java/org/mozilla/fenix/settings/wallpaper/ExtensionsTest.kt

```diff
@@ -0,0 +1,93 @@
+        val seasonalCollection = getSeasonalCollection("finally fall")
+        val classicFirefoxWallpapers = (0..5).map { generateClassicFirefoxWallpaper("firefox$it") }
+        val seasonalWallpapers = (0..5).map { generateSeasonalWallpaperCollection("${seasonalCollection.name}$it", seasonalCollection.name) }
+        val allWallpapers = listOf(Wallpaper.Default) + classicFirefoxWallpapers + seasonalWallpapers
+
+        val result = allWallpapers.groupByDisplayableCollection()
+
+        assertEquals(2, result.size)
+        assertEquals(listOf(Wallpaper.Default) + classicFirefoxWallpapers, result[classicCollection])
+        assertEquals(seasonalWallpapers, result[seasonalCollection])
+    }
+
+    @Test
+    fun `GIVEN no wallpapers but the default WHEN grouped by collection THEN the default will still be present`() {
+        val result = listOf(Wallpaper.Default).groupByDisplayableCollection()
+
+        assertEquals(1, result.size)
+        assertEquals(listOf(Wallpaper.Default), result[Wallpaper.DefaultCollection])
+    }
+
+    @Test
+    fun `GIVEN wallpapers with thumbnails that have not downloaded WHEN grouped by collection THEN wallpapers without thumbnails will not be included`() {
+        val seasonalCollection = getSeasonalCollection("finally fall")
+        val classicFirefoxWallpapers = (0..5).map { generateClassicFirefoxWallpaper("firefox$it") }
+        val downloadedSeasonalWallpapers = (0..5).map { generateSeasonalWallpaperCollection("${seasonalCollection.name}$it", seasonalCollection.name) }
+        val nonDownloadedSeasonalWallpapers = (0..5).map {
+            generateSeasonalWallpaperCollection(
+                "${seasonalCollection.name}$it",
+                seasonalCollection.name,
+                Wallpaper.ImageFileState.Error,
+            )
+        }
+        val allWallpapers = listOf(Wallpaper.Default) + classicFirefoxWallpapers + downloadedSeasonalWallpapers + nonDownloadedSeasonalWallpapers
+
+        val result = allWallpapers.groupByDisplayableCollection()
+
+        assertEquals(2, result.size)
+        assertEquals(listOf(Wallpaper.Default) + classicFirefoxWallpapers, result[classicCollection])
+        assertEquals(downloadedSeasonalWallpapers, result[seasonalCollection])
+    }
+
+    private val classicCollection = Wallpaper.Collection(
+        name = "classic-firefox",
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+    )
+
+    private fun generateClassicFirefoxWallpaper(name: String) = Wallpaper(
+        name = name,
+        textColor = 0L,
+        cardColor = 0L,
+        thumbnailFileState = Wallpaper.ImageFileState.Downloaded,
+        assetsFileState = Wallpaper.ImageFileState.Downloaded,
+        collection = classicCollection,
+    )
+
+    private fun getSeasonalCollection(name: String) = Wallpaper.Collection(
+        name = name,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
```

Comment:
Can you add all the named parameters? This will futureproof us if we ever have to refactor any of these parameters in `Wallpaper.Collection`